lib/shopping_cart.rb

```diff
@@ -0,0 +1,53 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    {name: @name, capacity: @capacity}
+  end
+
+  def total_number_of_products
+    @products.reduce(0) do |sum, product|
+      sum + product.quantity
+    end
+  end
+
+  def is_full?
```

Comment:
A tiny bit redundant, this boolean method can be one line if we only have
`total_number_of_products >= capacity` as this will always evaluate to true or false